src/internal.c

```diff
@@ -5649,13 +5649,14 @@ static int BuildFinished(WOLFSSL* ssl, Hashes* hashes, const byte* sender)
 #endif
 #ifndef NO_OLD_TLS
     if (!ssl->options.tls) {
-        if (BuildMD5(ssl, hashes, sender) != 0) {
+        ret = BuildMD5(ssl, hashes, sender);
+        if (ret != 0) {
         #ifdef WOLFSSL_SMALL_STACK
         #ifdef WOLFSSL_SHA384
             XFREE(sha384, NULL, DYNAMIC_TYPE_TMP_BUFFER);
         #endif
         #endif
-            return SSL_FATAL_ERROR;
+            return ret;
```

Comment:
Does "ssl->hsHashes->hashSha384" need to be restored here before returning?  Asking because if BuildTlsFinished() fails above, we restored it first before returning.